### PR TITLE
Cleaner message printing

### PR DESCRIPTION
**Pull request by AltGr**
Aerates the printing of messages, removes extraneous noise, and more clearly attaches the various information tied to a given message.

### CODE REVIEW

**Comment by AltGr** [comment]
Before:
image
After:
image

**Comment by denismerigoux** [comment]
I'm telling you you're reimplementing https://gitlab.com/alistair.obrien/grace at this point :)

File: compiler/catala_utils/message.ml [hunk 1]
```diff
@@ -164,28 +164,117 @@ module Content = struct
...
+            pos.pos_message;
+          Pos.format_loc_text ppf pos.pos
+        | MainMessage msg ->
+          Format.fprintf ppf "@[<hov 2>[%t] %t@]" (pp_marker target) msg
+        | Outcome msg ->
+          Format.fprintf ppf "@[<hov>[%t]@ %t@]" (pp_marker target) msg
+        | Suggestion suggestions_list -> Suggestions.format ppf suggestions_list)
+      ppf content;
+    Format.pp_close_box ppf ();
+    Format.pp_print_newline ppf ()
```
Comment (denismerigoux) [inline compiler/catala_utils/message.ml:184]: Why two different message formats (`basic` and `fancy`)?

File: compiler/catala_utils/message.ml [hunk 2]
```diff
@@ -164,28 +164,117 @@ module Content = struct
...
-          | Outcome msg ->
-            Format.fprintf ppf "@[<hov>%t@ %t@]" (pp_marker target) msg
-          | Suggestion suggestions_list ->
-            Suggestions.format ppf suggestions_list)
-        ppf content;
-      Format.pp_close_box ppf ();
-      Format.pp_print_newline ppf ()
+      match target with
+      | Debug | Log -> basic_msg ppf target content
+      | Result | Warning | Error -> fancy_msg ppf target content)
```
Comment (denismerigoux) [inline compiler/catala_utils/message.ml:277]: Interesting, what's the rationale for printing debug and log messages in a degraded format?

File: compiler/catala_utils/pos.mli [hunk 3]
```diff
@@ -59,5 +59,13 @@ val format_loc_text : Format.formatter -> t -> unit
 (** Open the file corresponding to the position and retrieves the text concerned
     by the position *)
 
+val format_loc_text_parts :
+  t ->
+  (Format.formatter -> unit)
+  * (Format.formatter -> unit)
+  * (Format.formatter -> unit) option
+(** Like [format_loc_text], but returns the printing functions in 3 separate
```
Comment (denismerigoux) [inline compiler/catala_utils/pos.mli:67]: If it's not used anymore, can you delete the old `format_loc_text`?

File: tests/arithmetic/good/priorities.catala_en [hunk 4]
```diff
@@ -16,15 +16,21 @@ scope A:
 
 ```catala-test-inline
 $ catala Typecheck --check-invariants
-[RESULT] All invariant checks passed
-[RESULT] Typechecking successful!
+┌─[RESULT]─
+│All invariant checks passed
```
Comment (denismerigoux) [inline tests/arithmetic/good/priorities.catala_en:20]: Missing a space between the left line and the beginning of the message.

File: tests/arithmetic/good/priorities.catala_en [hunk 5]
```diff
@@ -16,15 +16,21 @@ scope A:
...
 $ catala test-scope A
-[RESULT] Computation successful! Results:
-[RESULT] w = 0
-[RESULT] x = 4
-[RESULT] y = 4
-[RESULT] z = 390.0
+┌─[RESULT]─
+│Computation successful! Results:
+│w = 0
+│x = 4
```
Comment (denismerigoux) [inline tests/arithmetic/good/priorities.catala_en:32]: Ditto space missing

**Review by denismerigoux — APPROVED** [review]
Very beautiful @AltGr!

File: compiler/catala_utils/message.ml (hunk 1 above)
Comment (AltGr) [reply compiler/catala_utils/message.ml:184]: `basic` is still used for logs and debugs

File: compiler/catala_utils/message.ml (hunk 2 above)
Comment (AltGr) [reply compiler/catala_utils/message.ml:277]: Better answered with a screenshot:
image

---
If we were to use the "fancy" printer:
image

Comment (denismerigoux) [reply compiler/catala_utils/message.ml:277]: I see, that's a reasonable decision, thanks!

**Comment by AltGr** [comment]
> I'm telling you you're reimplementing https://gitlab.com/alistair.obrien/grace at this point :)

Hm not really, this is only our "general" message formatting, I didn't change how we point to specific locations inside the original sources.

File: compiler/catala_utils/pos.mli (hunk 3 above)
Comment (AltGr) [reply compiler/catala_utils/pos.mli:67]: It's still used for now, when displaying traces for example

**Pull request merged**